src/evict/evict.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+// TODO - prototypes.py overwrites this file and removes all of my beautiful dependency include work.
+// Need to update s_all to stop doing this
+#include "../include/wt_prelude.h"
+
+#include "../support/support.h"         // Required dependency
```

Comment:
This is the list of dependencies that evict needs to compile, but many - if not most - aren't actually needed by eviction.

By my count we have:
- 8 "true" dependencies
- 6 dependencies that can be removed if we stop using a single function
- 9 transitive dependencies. These can be ignored. They wouldn't be here if the rest of the system also #included its dependencies directly.

The includes are limited to the public header, so we only need to do them once for all .c files. I think it's also useful since it highlights dependencies we could easily break and potentially problematic dependency cycles.